mlir/lib/Dialect/Tensor/Transforms/FoldTensorSubsetOps.cpp

```diff
@@ -154,6 +166,25 @@ LogicalResult InsertSliceOfTransferWriteOpFolder::matchAndRewrite(
   return success();
 }
 
+bool InsertSliceOfTransferWriteOpFolder::doesTransferWriteCoverInsertSlice(
+    vector::TransferWriteOp writeOp, tensor::InsertSliceOp insertSliceOp,
+    MLIRContext *context) {
+  auto destType = cast<ShapedType>(writeOp.getOperand(0).getType());
+  auto insertSliceType = insertSliceOp.getSourceType();
+
+  if (destType.hasStaticShape() && insertSliceType.hasStaticShape()) {
+    for (int64_t d = 0, e = insertSliceType.getRank(); d < e; ++d) {
+      if (destType.getDimSize(d) != insertSliceType.getDimSize(d))
+        return false;
+    }
```

Comment:
I think this could just do something like:
```suggestion
llvm::equal(writeOp.getVectorType().getShape(), writeOp.getShapedType().getShape());
```

Reply:
Done.